flink-runtime/src/test/java/org/apache/flink/runtime/taskmanager/TaskTest.java

```diff
@@ -855,7 +862,8 @@ public void testFatalErrorOnCanceling() throws Exception {
 			task.cancelExecution();
 
 			// wait for the notification of notifyFatalError
-			taskManagerActions.latch.await();
+			final Throwable fatalError = fatalErrorFuture.join();
+			assertThat(fatalError, instanceOf(fatalErrorType));
 		} catch (Throwable t) {
 			fail("No exception is expected to be thrown by fatal error handling");
```

Comment:
Explicitly failing the test seems unnecessarily verbose.

Reply:
True. I will refactor the test case.